packages/aws-cdk/test/api/hotswap/hotswap-test-setup.ts

```diff
@@ -38,16 +61,29 @@ export function cdkStackArtifactOf(testStackArtifact: Partial<TestStackArtifact>
   });
 }
 
-export function pushStackResourceSummaries(...items: CloudFormation.StackResourceSummary[]) {
+export function pushStackResourceSummaries(...items: AWS.CloudFormation.StackResourceSummary[]) {
   currentCfnStackResources.push(...items);
 }
 
+export function pushNestedStackResourceSummaries(stackName: string, ...items: AWS.CloudFormation.StackResourceSummary[]) {
+  if (!currentNestedCfnStackResources[stackName]) {
+    currentNestedCfnStackResources[stackName] = [];
+  }
+
+  currentNestedCfnStackResources[stackName].push(...items);
+}
+
 export function setCurrentCfnStackTemplate(template: Template) {
   const templateDeepCopy = JSON.parse(JSON.stringify(template)); // deep copy the template, so our tests can mutate one template instead of creating two
   currentCfnStack.setTemplate(templateDeepCopy);
 }
 
-export function stackSummaryOf(logicalId: string, resourceType: string, physicalResourceId: string): CloudFormation.StackResourceSummary {
+export function addTemplateToCloudFormationLookupMock(stackArtifact: CloudFormationStackArtifact) {
+  const templateDeepCopy = JSON.parse(JSON.stringify(stackArtifact.template));
```

Comment:
Probably a comment here is warranted too, like in `setCurrentCfnStackTemplate()`.